File: frontend/lib/dtos/request/create_task.dto.dart
class CreateTaskDto {
  final String assignedTo;
  final List<String> blockedBy;
  final List<String> tags;
  final String title;
  final String description;
  final String priority;
  final String status;
  final num? expectedHours;

  CreateTaskDto({
    this.assignedTo = '',
    this.blockedBy = const [],
    this.tags = const [],
    this.title = '',
    this.description = '',
    this.priority = 'NORMAL',
    this.status = 'TODO',
    this.expectedHours,
  });

  Map<String, dynamic> get json {
    return {
      'assignedTo': assignedTo,
      'blockedBy': blockedBy,
      'tags': tags,
      'title': title,
      'description': description,
      'priority': priority,
      'status': status,
      'expectedHours': expectedHours,
    };
  }
}

File: backend/src/main/kotlin/taskmanager/backend/dtos/request/CreateTaskDto.kt
package taskmanager.backend.dtos.request

import taskmanager.backend.enums.Priority
import taskmanager.backend.enums.Status

data class CreateTaskDto(
    val assignedTo: String,
    val blockedBy: List<String> = emptyList(),
    val tags: List<String> = emptyList(),
    val title: String,
    val description: String,
    val priority: String,
    val status: String,
    val expectedHours: Int?
) {

    fun validate() {
        Priority.valueOf(priority)
        Status.valueOf(status)
    }
}